src/spyOn.ts

```diff
@@ -28,12 +28,16 @@ export function spyOn<Obj extends object, Method extends Methods<Obj>>(
   obj: Obj,
   methodName: Method,
   mock?: Obj[Method]
-): Spy<Parameters<Obj[Method]>, ReturnType<Obj[Method]>>
+): Obj[Method] extends (...args: Array<any>) => any
```

Comment:
I think we can use `infer` here:
```
(...args: infer Args) => infer Returns
```

to pass them to `Spy`

Reply:
Don't really get you on this. Could you elaborate?

Reply:
Not really familiar with infer probably

Reply:
```
O[M] extends (...args: infer A) => R ? Spy<A, R> : never
```

Reply:
we also had to infer R